src/services/csharp.js

```diff
@@ -0,0 +1,719 @@
+        }
+        else {
+          testCaseLines.push(new Line('String appURL = testApp.getAppUrl();'))
+          testCaseLines.push(new Line(
+            `AppiumOptions capabilities = Config.${desiredCapsMethodName}(appURL);`))
+        }
+        testCaseLines.push(new Line('testApp.FindOnlineDevice(capabilities);'))
+        testCaseLines.push(new Line(`testApp.Setup(capabilities, ${retinaScale});`))
+        testCaseLines.push(new Line('testApp.RunTest();'))
+      }
+
+      testCaseLines.push(new Line('}', -1))
+      testCaseLines.push(new Line(''))
+    }
+
+    return testCaseLines
+  }
+
+  /**
+   * Generates an array of lines for a test script based on the provided test steps.
+   * @param {Object} options - The options object.
+   * @param {Array} options.testSteps - The array of test steps to generate the test
+   * script lines from.
+   * @returns {Object} An object containing the generated lines and resource files.
+   */
+  _generateTestScriptLines({testSteps}) {
+    const lines = []
+    const resourceFiles = {}
+    const locatorVarNames = new Set()
+
+    for (const step of testSteps) {
+      const {
+        id, context, actionJson, selectorConfigurations, isOnKeyboard, findingElementTimeout
+      } = step
+      if (!actionJson) continue
+
+      let action
+      try {
+        action = JSON.parse(actionJson)
+      }
+      catch (error) {
+        console.error(`Cannot parse actionJson: ${actionJson}`)
+        throw error
+      }
+
+      const actionCommand = get(action, 'command')
+      const hasSelector = selectorConfigurations && selectorConfigurations.length
+
+      lines.push(new Line(''))
+      lines.push(new Line(`SetCurrentCommandId(${step.id});`))
+
+      let locatorVarName, rawLocatorVarName
+      if (hasSelector) {
+        locatorVarName = this._getLocatorVarName(step, locatorVarNames)
+        rawLocatorVarName = locatorVarName.replace(LOCATOR_VAR_NAME_PREFIX, '')
+        const locatorCode = this._getLocatorCode({step, locatorVarName})
+        lines.push(...locatorCode)
+      }
+
+      switch (actionCommand) {
+        case 'activateApp': {
+          const {appPackage} = action
+          lines.push(new Line(`driver.activateApp("${appPackage}");`))
+        } break
+
+        case 'touchOnElement': {
+          const {x, y} = action
+          if (context === CONTEXTS.NATIVE) {
+            const elementVarName = `element${rawLocatorVarName}`
+            // eslint-disable-next-line max-len
+            lines.push(new Line(`AppiumWebElement ${elementVarName} = FindElementBy(${findingElementTimeout}, ${locatorVarName});`))
+            // eslint-disable-next-line max-len
+            lines.push(new Line(`TouchOnElementByType(${elementVarName}, ${x}, ${y});`))
+            //
+            try {
+              const {value} = action
+              if (value != undefined) {
+                lines.push(new Line(`${elementVarName}.SendKeys(${this._getString(value)});`))
```

Comment:
on "touchOnElement" step we cannot have "value" field for sendKeys command, it should be a separated step